rilldata/rill
# YAML editor; integration into `/dashboard/<name>/edit`

## Checklist

## Summary
#### Issue addressed: 

Integrates a new YAML editor, which is essentially a handful of codemirror plugins. This work isn't perfect but it's a big step in the right direction. 

Concretely, this PR:
- adds in `web-common/src/components/editor`, which maintains the codemirror plugins.
- adds a _line status system_, a means of adding line-specific errors and having the line color change. This system includes a custom gutter to display an error icon.
- adds in indent guides a la VSCode.
- adds a _metrics editor_ in YAML, which (1) utilizes a simple inspector (2) places file errors additionally as a bottom component that slides up and (3) implements a placeholder that handles guiding the user to either bootstrap off an existing model or give a skeleton.

## Steps to Verify
- use the "autogenerate" functionality from anywhere in the app
- try to make a blank dashboard and go through the different provided paths in the placeholder
- edit the config and see the changes in the dash
